### PR TITLE
Use nginx to route requests from port 8003 to connector builder

### PR DESCRIPTION
**Pull request by girarda**
## What
I misunderstood how to use the proxy to forward requests to the connector-builder-server in my [original PR](https://github.com/airbytehq/airbyte/pull/18410).

This use nginx to route requests sent to localhost:8003 to the connector-builder-server:80.

Everything is still commented out so this PR won't affect production.

## How
* Add a server block listening to port 8003 to the nginx-auth config templates. Requests are forwarded to http://airbyte-connector-builder-server:80
* Add an env var to the proxy's Dockefile so it knows where the connector-builder-server lives
* Update the proxy run script to replace the new env var in the nginx templates
* Update the proxy test script to pass in the new env var
* Update docker-compose to expose port 80 on the connector-builder-server
* Update kube/resources/connector-builder-server.yaml to expose port 80 on the connector-builder-server

## Recommended reading order
1. `airbyte-proxy/nginx-auth.conf.template`
2. `airbyte-proxy/nginx-no-auth.conf.template`
3. `airbyte-proxy/Dockerfile`
4. `airbyte-proxy/run.sh`
5. `airbyte-proxy/test.sh`
6. `docker-compose.yaml`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-proxy/run.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ else
   TEMPLATE_PATH="/etc/nginx/templates/nginx-auth.conf.template"
 fi
 
-envsubst '${PROXY_PASS_WEB} ${PROXY_PASS_API} ${PROXY_PASS_RESOLVER}' < $TEMPLATE_PATH > /etc/nginx/nginx.conf
+envsubst '${PROXY_PASS_WEB} ${PROXY_PASS_API} ${CONNECTOR_BUILDER_SERVER_API} ${PROXY_PASS_RESOLVER}' < $TEMPLATE_PATH > /etc/nginx/nginx.conf
```
Comment (girarda) [inline airbyte-proxy/run.sh:19]: search and replace the patterns with the value of the corresponding bash variable
https://www.baeldung.com/linux/envsubst-command#:~:text=The%20envsubst%20command%20searches%20the,envsubst%20recognizes%20only%20exported%20variables.

File: airbyte-proxy/Dockerfile [hunk 2]
```diff
@@ -12,6 +12,7 @@ RUN apt-get update -y && apt-get install -y apache2-utils && rm -rf /var/lib/apt
 # This variable can be used to update the destintion containers that Nginx proxies to.
 ENV PROXY_PASS_WEB "http://airbyte-webapp:80"
 ENV PROXY_PASS_API "http://airbyte-server:8001"
+ENV CONNECTOR_BUILDER_SERVER_API "http://airbyte-connector-builder-server:80"
```
Comment (girarda) [inline airbyte-proxy/Dockerfile:15]: requests will come on port 8003, but will be forwarded to the port 80 on the connector-builder-server

File: docker-compose.yaml [hunk 3]
```diff
@@ -210,24 +210,24 @@ services:
...
-  #    - AIRBYTE_VERSION=${VERSION}
-  #  networks:
-  #    - airbyte_internal
+  airbyte-connector-builder-server: #FIXME: Uncomment this block when enabling airbyte-connector-builder
+    image: airbyte/connector-builder-server:${VERSION}
+    logging: *default-logging
+    container_name: airbyte-connector-builder-server
+    restart: unless-stopped
+    ports:
+      - 80
```
Comment (girarda) [inline docker-compose.yaml:219]: expose port 80 because the server runs on port 80

File: docker-compose.yaml [hunk 4]
```diff
@@ -210,24 +210,24 @@ services:
...
+    networks:
+      - airbyte_internal
   airbyte-proxy:
     image: airbyte/proxy:${VERSION}
     container_name: airbyte-proxy
     ports:
       - 8000:8000
       - 8001:8001
-      #- 8003:8003 FIXME: Uncomment this when enabling airbyte-connector-builder
+      - 8003:8003 # FIXME: Uncomment this when enabling airbyte-connector-builder
```
Comment (girarda) [inline docker-compose.yaml:230]: forward the proxy's port 8003 to localhost:8003

File: airbyte-proxy/nginx-no-auth.conf.template [hunk 5]
```diff
@@ -24,4 +24,16 @@ http {
...
+
+    server {
+      listen 8003;
+
+      location / {
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+
+        proxy_pass "${CONNECTOR_BUILDER_SERVER_API}";
```
Comment (girarda) [inline airbyte-proxy/nginx-no-auth.conf.template:36]: forward to "http://airbyte-connector-builder-server:80" as defined in https://github.com/airbytehq/airbyte/pull/19264/files#diff-8a4136717271f90d70120fef7bb8904d26f3fbcfe7520dbd39fa0b8723d8548eR15

File: docker-compose.yaml [hunk 6]
```diff
@@ -211,23 +211,23 @@ services:
...
-  #  environment:
-  #    - AIRBYTE_VERSION=${VERSION}
-  #  networks:
-  #    - airbyte_internal
+  #image: airbyte/connector-builder-server:${VERSION}
+  #logging: *default-logging
+  #container_name: airbyte-connector-builder-server
+  #restart: unless-stopped
+  #ports:
+  #  - 80
```
Comment (girarda) [inline docker-compose.yaml:219]: expose port 80 because the server runs on port 80

File: docker-compose.yaml [hunk 7]
```diff
@@ -211,23 +211,23 @@ services:
...
+  #networks:
+  #  - airbyte_internal
   airbyte-proxy:
     image: airbyte/proxy:${VERSION}
     container_name: airbyte-proxy
     ports:
       - 8000:8000
       - 8001:8001
-      #- 8003:8003 FIXME: Uncomment this when enabling airbyte-connector-builder
+      #- 8003:8003 # FIXME: Uncomment this when enabling airbyte-connector-builder
```
Comment (girarda) [inline docker-compose.yaml:230]: forward the proxy's port 8003 to localhost:8003

File: airbyte-proxy/nginx-auth.conf.template [hunk 8]
```diff
@@ -42,4 +42,26 @@ http {
...
+
+#    location / {
+#      proxy_set_header Host $host;
+#      proxy_set_header X-Real-IP $remote_addr;
+#      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+
+#      auth_basic "Welcome to Airbyte";
+#      auth_basic_user_file /etc/nginx/.htpasswd;
+
+#      proxy_pass "${CONNECTOR_BUILDER_SERVER_API}";
```
Comment (girarda) [inline airbyte-proxy/nginx-auth.conf.template:57]: forward to "[http://airbyte-connector-builder-server:80](http://airbyte-connector-builder-server/)" as defined in https://github.com/airbytehq/airbyte/pull/19264/files#diff-8a4136717271f90d70120fef7bb8904d26f3fbcfe7520dbd39fa0b8723d8548eR15

File: kube/resources/connector-builder-server.yaml [hunk 9]
```diff
@@ -1,3 +1,15 @@
+apiVersion: v1
```
Comment (girarda) [inline kube/resources/connector-builder-server.yaml:1]: this file is not imported in the kube deploys (https://github.com/airbytehq/airbyte/blob/master/kube/resources/kustomization.yaml#L7)

File: airbyte-proxy/test.sh [hunk 10]
```diff
@@ -11,7 +11,7 @@ VERSION="${VERSION:-dev}" # defaults to "dev", otherwise it is set by environmen
 echo "testing with proxy container airbyte/proxy:$VERSION"
 
 function start_container () {
-  CMD="docker run -d -p $PORT:8000 --env BASIC_AUTH_USERNAME=$1 --env BASIC_AUTH_PASSWORD=$2 --env PROXY_PASS_WEB=http://localhost --env PROXY_PASS_API=http://localhost --name $NAME airbyte/proxy:$VERSION"
+  CMD="docker run -d -p $PORT:8000 --env BASIC_AUTH_USERNAME=$1 --env BASIC_AUTH_PASSWORD=$2 --env PROXY_PASS_WEB=http://localhost --env PROXY_PASS_API=http://localhost --env CONNECTOR_BUILDER_SERVER_API=http://localhost --name $NAME airbyte/proxy:$VERSION"
```
Comment (lmossman) [inline airbyte-proxy/test.sh:14]: Just curious, why are these just `http://localhost` instead of `http://localhost:PORT`?

**Review by lmossman — APPROVED** [review]
One small question but otherwise these changes LGTM!

File: airbyte-proxy/test.sh (hunk 10 above)
Comment (girarda) [reply airbyte-proxy/test.sh:14]: this test is not spinning up the whole platform to test the routing. It's only testing that unauthenticated requests fail with a 401 error and that authenticated do not result in a 401 error

**Review by brianjlai — APPROVED** [review]

**Pull request closed without merging**